src/Libraries/Storage/Adapters/GoogleDrive/GoogleDriveApp.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Quantum\Libraries\Storage\Adapters\GoogleDrive;
+
+use Quantum\Libraries\Curl\HttpClient;
+use Quantum\Http\Response;
+use Exception;
+
+class GoogleDriveApp
+{
+    /**
+     * Authorization URL
+     */
+    const AUTH_URL = 'https://accounts.google.com/o/oauth2/v2/auth';
+
+    /**
+     * Authorization scope
+     */
+    const AUTH_SCOPE = 'https://www.googleapis.com/auth/drive';
+
+    /**
+     * Token URL
+     */
+    const AUTH_TOKEN_URL = 'https://oauth2.googleapis.com/token';
+
+    /**
+     * URL for file metadata operations
+     */
+    const FILE_METADATA_URL = 'https://www.googleapis.com/drive/v3/files';
+
+    /**
+     * URL for file media operations
+     */
+    const FILE_MEDIA_URL = 'https://www.googleapis.com/upload/drive/v3/files';
+
+    /**
+     * Folder mimetype
+     */
+    const FOLDER_MIMETYPE = 'application/vnd.google-apps.folder';
+
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var string
+     */
+    private $appKey = null;
+
+    /**
+     * @var string
+     */
+    private $appSecret = null;
+
+    /**
+     * @var TokenServiceInterface
+     */
+    private $tokenService = null;
+
+    /**
+     * GoogleDriveApp constructor
+     * @param string $appKey
+     * @param string $appSecret
+     * @param TokenServiceInterface $tokenService
+     * @param HttpClient $httpClient
+     */
+    public function __construct(string $appKey, string $appSecret, TokenServiceInterface $tokenService, HttpClient $httpClient)
+    {
+        $this->appKey = $appKey;
+        $this->appSecret = $appSecret;
+        $this->tokenService = $tokenService;
+        $this->httpClient = $httpClient;
+    }
+
+    public function getAuthUrl($redirectUrl, $accessType = "offline"): string
+    {
+        $params = [
+            'client_id' => $this->appKey,
+            'response_type' => 'code',
+            'state' => csrf_token(),
+            'scope' => self::AUTH_SCOPE,
+            'redirect_uri' => $redirectUrl,
+            'access_type' => $accessType,
+        ];
+
+        return self::AUTH_URL . '?' . http_build_query($params, '', '&');
+    }
+
+    public function getAndSaveAccessToken($code, $redirectUrl = '', $byRefresh = false): object
```

Comment:
Name this function to `fetchTokens()` to correspond to DroboxApp naming convention